apps/dashboard/src/components/subscribers/subscriber-drawer.tsx

```diff
@@ -0,0 +1,62 @@
+import { motion } from 'motion/react';
+import { Sheet, SheetContentBase, SheetDescription, SheetPortal, SheetTitle } from '../primitives/sheet';
+import { VisuallyHidden } from '../primitives/visually-hidden';
+import { useNavigate } from 'react-router-dom';
+import { PropsWithChildren } from 'react';
+
+const transitionSetting = { duration: 0.4 };
+
+type SubscriberDrawerProps = PropsWithChildren<{
+  open: boolean;
+}>;
+
+export function SubscriberDrawer({ children, open }: SubscriberDrawerProps) {
+  const navigate = useNavigate();
+
+  const handleCloseSheet = () => {
+    navigate(-1);
+  };
+
+  return (
+    <Sheet modal={false} open={open}>
+      <motion.div
+        initial={{
+          opacity: 0,
+        }}
+        animate={{
+          opacity: 1,
+        }}
+        exit={{
+          opacity: 0,
+        }}
+        className="fixed inset-0 z-50 h-screen w-screen bg-black/20"
+        transition={transitionSetting}
+      />
+      <SheetPortal>
+        <SheetContentBase asChild onInteractOutside={handleCloseSheet} onEscapeKeyDown={handleCloseSheet}>
+          <motion.div
```

Comment:
Since this Sheet is reusable across multiple screens (Create workflows, create environment, etc...) the appearance and exit animation should be part of the Sheet and not redefined every time we use it.